commands/actions/publish.go

```diff
@@ -317,12 +338,23 @@ func mustGetSource(sourcesDir string, locator string) string {
 	return content
 }
 
-func mustValidate(r *rest.Rest, actions *actionsModel.ProjectActions, sources map[string]string, projectSlug string) {
-	// TODO(marko): Send logic & dependencies version in validate request
+func mustValidate(
+	r *rest.Rest,
+	actions *actionsModel.ProjectActions,
+	sources map[string]string,
+	projectSlug string,
+) (logicExist bool, dependenciesExist bool) {
 	request := actions2.ValidateRequest{
-		Actions: actions.ToRequest(sources),
+		Actions:             actions.ToRequest(sources),
+		LogicVersion:        nil,
+		DependenciesVersion: nil,
 	}
 
+	_, request.LogicVersion = util.MustZipAndHashDir(outDir, srcPathInZip, zipLimitBytes)
+
+	dependenciesDir := filepath.Join(actions.Sources, typescript.NodeModulesDir)
+	_, request.DependenciesVersion = util.ZipAndHashDir(dependenciesDir, nodeModulesPathInZip, zipLimitBytes)
```

Comment:
Why in one place we're using must and in the other just ZipAndHash?

Reply:
Also, won't there already be a dir check in `MustZipDir`?

Reply:
`MustZipAndHashDir` is used for the action source code which must exist and `ZipAndHashDir` is used for the node_modules folder, which doesn't need to be present. 
`MustZipDir` is checking if dir exists and exiting if it doesn't, which is not suitable for node_modules.
Do you have any hint for refactoring these so they are usable in both cases?